extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -140,9 +140,13 @@ void gatherSecurityChecks(BuildProducer<BeanRegistrarBuildItem> beanRegistrars,
             List<AdditionalSecurityCheckBuildItem> additionalSecurityChecks, SecurityBuildTimeConfig config) {
         classPredicate.produce(new ApplicationClassPredicateBuildItem(new SecurityCheckStorage.AppPredicate()));
 
-        Set<ClassInfo> additionalSecured = new HashSet<>();
+        final Map<DotName, ClassInfo> additionalSecured = new HashMap<>();
         for (AdditionalSecuredClassesBuildIem securedClasses : additionalSecuredClasses) {
-            additionalSecured.addAll(securedClasses.additionalSecuredClasses);
+            securedClasses.additionalSecuredClasses.forEach(c -> {
+                if (!additionalSecured.containsKey(c.name())) {
+                    additionalSecured.put(c.name(), c);
+                }
+            });
```

Comment:
I don't think it's worth changing it but please when using a for loop is simpler than using a lambda, please let's do the former.

Using lambdas in build time code is allowed but maybe let's only use them when they bring something to the plate?